Library/include/CSP/Systems/Log/LogSystem.h

```diff
@@ -71,19 +71,27 @@ class CSP_API CSP_NO_DISPOSE LogSystem
 	CSP_EVENT void SetEndMarkerCallback(EndMarkerCallbackHandler InEndCallback);
 
 	/// @brief Set the verbosity of logging for a system-wide level.
-	CSP_NO_EXPORT void SetSystemLevel(const csp::systems::LogLevel InSystemLevel);
+	/// @param InSystemLevel The level to set the system logging to.
+	void SetSystemLevel(const csp::systems::LogLevel InSystemLevel);
 
 	/// @brief Retreive the log verbosity level.
-	CSP_NO_EXPORT csp::systems::LogLevel GetSystemLevel();
+	csp::systems::LogLevel GetSystemLevel();
 
 	/// @brief Check if we currently log a specified log verbosity level.
-	CSP_NO_EXPORT bool LoggingEnabled(const csp::systems::LogLevel Level);
+	/// @param Level The level to check.
+	bool LoggingEnabled(const csp::systems::LogLevel Level);
 
 	/// @brief Log a message at a specific verbosity level.
+	/// @param Level The level to log this message at.
+	/// @param InMessage The message to be logged.
 	CSP_NO_EXPORT void LogMsg(const csp::systems::LogLevel Level, const csp::common::String& InMessage);
```

Comment:
Any reason to keep this as NOEXPORT?

Reply:
For this and `LogEvent` it seems like it would be a significant change to do as part of this ticket as it would open up the log system for clients to use for their own logging too. I don't really have an opinion on whether that's good/bad/otherwise, just that it's a fair bit beyond the scope of this ticket.

Reply:
Ok

Reply:
These functions are already exposed to anything that uses the C++ DLL directly, like Unreal

Reply:
Should we remove all `CSP_NO_EXPORT` from this so? It's quite a change to the scope of the ticket but if we're happy to do so then I'm happy make the change. @MAG-SamBirley, happy to update the ticket?

Reply:
I'm ok treating it as out of scope for this change, but @MAG-pw could you create another one for us to get rid of the other NOEXPORTs?

Reply:
Will do.

Reply:
@MAG-SamBirley ticket created an assigned to you for prioritization.